### DISCUSSION TITLE
BUILD FAILED (FreeBSD 13.1 using ruby-build 20230309-1-gf3c1ccc)

### DISCUSSION THREAD

**Discussion by duncan-bayne** [rbenv/ruby-build#2168] [Build failures] [answered]
I'm trying to build Ruby 3.2.1 on FreeBSD 13.1, using rbenv and ruby-build through the [Sourcehut build system](https://man.sr.ht/builds.sr.ht/), but it's failing:

```
*** Following extensions are not compiled:
fiddle:
	Could not be configured. It will not be installed.
	/tmp/ruby-build.20230320233100.1016.N3xMg3/ruby-3.2.1/ext/fiddle/extconf.rb:73: missing libffi. Please install libffi or use --with-libffi-source-dir with libffi source location.
	Check ext/fiddle/mkmf.log for more details.
psych:
	Could not be configured. It will not be installed.
	Check ext/psych/mkmf.log for more details.
*** Fix the problems, then remove these directories and try again if you want.
gmake[1]: Leaving directory '/tmp/ruby-build.20230320233100.1016.N3xMg3/ruby-3.2.1'
Generating RDoc documentation
/tmp/ruby-build.20230320233100.1016.N3xMg3/ruby-3.2.1/lib/yaml.rb:3: warning: It seems your ruby installation is missing psych (for YAML output).
To eliminate this warning, please install libyaml and reinstall your ruby.
uh-oh! RDoc had a problem:
cannot load such file -- psych

run with --debug for full backtrace
gmake: *** [uncommon.mk:598: rdoc] Error 1
```

However, libyaml is definitely installed (`libyaml: 0.2.5` from the FreeBSD package repository).

[ruby-build.log](https://github.com/rbenv/ruby-build/files/11023874/ruby-build.log)

**Comment by duncan-bayne** [comment]
This has been reported on the FreeBSD bug tracker as well: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=269803

**Comment by jarmo** [comment]
Had the same problem on FreeBSD 12.2 and managed to build Ruby 3.2.2. I'm using [asdf](https://github.com/asdf-vm/asdf) and it's [asdf-ruby](https://github.com/asdf-vm/asdf-ruby) plugin, which uses `ruby-build` underneath.

Here are the commands which worked for me:
```
pkg install -y libyaml libffi
export RUBY_CONFIGURE_OPTS="--with-libyaml-dir=$(pkg info --prefix libyaml | cut -w -f 2) --with-libffi-dir=$(pkg info --prefix libffi | cut -w -f 2)"
asdf install ruby 3.2.2
```

`RUBY_CONFIGURE_OPTS` is for `ruby-build` and nothing to do with `asdf-ruby`.

It also has some commands/options regarding `libffi` which is not required for `psych`, but since I saw ruby-build warn about it then I included this too.

For me `--prefix` for `libyaml` and `libffi` is the same - `/usr/local`. Not sure why `ruby-build` doesn't find them correctly.

**Comment by jarmo** [comment]
Created issue https://github.com/rbenv/ruby-build/issues/2184

**Comment by jarmo** [comment]
@duncan-bayne fix merged into master branch.

**Comment by duncan-bayne** [comment]
Awesome, thanks @jarmo :heart:

**Comment by mislav** [comment, marked as answer]
[ruby-build 20230424](https://github.com/rbenv/ruby-build/releases/tag/v20230424) will now automatically link to libffi and libyaml installed via `pkg`. (Linking to openssl and libreadline was already present before.) Thank you @jarmo
